docs/versioned_docs/version-zenith/quickstart/822194-daggerize.mdx

```diff
@@ -0,0 +1,246 @@
+
+Try these steps by Daggerizing an example project - a [simple API server written in Go](https://github.com/kpenfound/greetings-api).
+
+### Fetch the project repository
+
+Fetch a copy of the project repository:
+
+```shell
+git clone https://github.com/kpenfound/greetings-api
+cd greetings-api
+```
+
+### Initialize a Dagger module
+
+Initialize a Dagger module at the root of the repository:
+
+```shell
+dagger init
+```
+
+You should see a new file at the root of your repository, named `dagger.json`. View its contents:
+
+```shell
+cat dagger.json
+```
+
+You should see the following:
+
+```json
+{
+  "name": "greetings-api",
+  "engineVersion": "v0.9.11"
+}
+```
+
+Finally, commit the file to your repository:
+
+```shell
+git add dagger.json
+git commit -m 'dagger init'
+```
+
+Congratulations! Your repository is now a Dagger Module.
+
+### Install a dependency
+
+ Now that your project is a Dagger Module, you can install other modules into it, as dependencies. A dependency is just a Dagger Module installed into another. As an example, install the `hello` module from earlier:
+
+```shell
+dagger install github.com/shykes/daggerverse/hello
+```
+
+Once the command completes, look at the `dagger.json` file again. You should see that a new dependency has been added, similar to that shown below:
+
+```json
+{
+  "name": "greetings-api",
+  "dependencies": [
+    {
+      "name": "hello",
+      "source": "github.com/shykes/daggerverse/hello@7884cbc1941f85a7c714fc7f1bcc5c148842c21e"
+    }
+  ],
+  "engineVersion": "v0.9.11"
+}
+```
+
+:::note
+The exact Git commit for the module version is recorded in `dagger.json`. Dagger enforces "version pinning", which guarantees that the module version you install is the one you'll be using.
+:::
+
+Go ahead and commit this change as well:
+
+```shell
+git add dagger.json
+git commit -m 'dagger install github.com/shykes/daggerverse/hello'
+```
+
+### Use the installed module
+
+4. Now that your module has its first dependency installed, you can start using it! To do so, simply refer to it by its short name - in this case, `hello`.
+
+List available functions:
+
+```shell
+dagger -m hello functions
+```
+
+You should see a familiar result:
+
+```shell
+Name    Description
+hello   Say hello to the world!
+```
+
+Call a function:
+
+```shell
+dagger -m hello call hello
+```
+
+Again, the output should be familiar:
+
+```shell
+hello, world!
+```
+
+### Daggerize a Go build
+
+The `hello` module is a good example to start with, but it doesn't have any functions to work with a Go project. To Daggerize the Go build for this example project, install the Go builder module from a previous section as another dependency:
+
+```shell
+dagger install github.com/kpenfound/dagger-modules/golang
+```
+
+Then, call the `Build()` function from the Go builder module to build the project and chain other function calls to export the built binary to the host:
+
+```shell
+dagger -m golang --proj . call \
```

Comment:
As discussed in https://discord.com/channels/707636530424053791/1120503349599543376/1209223864417124352, can you update this module to not use constructor args @kpenfound ? 

Note: this change will affect multiple examples in this quickstart.

Reply:
Updated so that the build function has an optional `--project` flag. The `--proj` constructor still works, but the `--project` flag can be used instead

Reply:
`dagger -m golang call build --project https://github.com/golang/example#master --args "-C","./hello"`

Reply:
Fixed thanks!